modules/library/referencing/src/main/java/org/geotools/referencing/operation/projection/Homolosine.java

```diff
@@ -0,0 +1,253 @@
+        double offset = computeOffset();
+        int i = 0;
+        double central_merid = 0;
+        Point2D p;
+        Point2D shift;
+        double thresh_map = LAT_THRESH; // spherical model
+
+        if (y >= 0) {
+            central_merids = CENTRAL_MERID_NORTH;
+            interruptions = new double[INTERRUP_NORTH.length];
+            for (int j = 0; j < INTERRUP_NORTH.length; j++)
+                interruptions[j] = sinu.transformNormalized(INTERRUP_NORTH[j], 0, null).getX();
+        } else {
+            central_merids = CENTRAL_MERID_SOUTH;
+            offset = -offset;
+            interruptions = new double[INTERRUP_SOUTH.length];
+            for (int j = 0; j < INTERRUP_SOUTH.length; j++)
+                interruptions[j] = sinu.transformNormalized(INTERRUP_SOUTH[j], 0, null).getX();
+        }
+
+        if (x > interruptions[interruptions.length - 1]) i = interruptions.length;
+        else while (x > interruptions[i]) i++;
+
+        central_merid = central_merids[i - 1];
+        shift = sinu.transformNormalized(central_merid, 0, null);
+
+        if (y > thresh_map || y < -thresh_map) { // Mollweide
+            p = moll.inverseTransformNormalized(x - shift.getX(), y + offset, ptDst);
+        } else { // Sinusoidal in spherical mode
+            p = new Point2D.Double((x - shift.getX()) / Math.cos(y), y);
+        }
+
+        p.setLocation(p.getX() + central_merid, p.getY());
+
+        if (ptDst != null) {
+            ptDst.setLocation(p.getX(), p.getY());
+            return ptDst;
+        } else {
+            return p;
+        }
+    }
+
+    //////////////////////////////////////////////////////////////////////////////////////////
+    //////////////////////////////////////////////////////////////////////////////////////////
+    ////////                                                                          ////////
+    ////////                                 PROVIDERS                                ////////
+    ////////                                                                          ////////
+    //////////////////////////////////////////////////////////////////////////////////////////
+    //////////////////////////////////////////////////////////////////////////////////////////
+
+    /**
+     * The {@linkplain org.geotools.referencing.operation.MathTransformProvider math transform
+     * provider} for the Homolosine projection (not part of the EPSG database).
+     *
+     * @since 22.x
+     * @author Luís M. de Sousa
+     * @see org.geotools.referencing.operation.DefaultMathTransformFactory
+     */
+    public static class Provider extends AbstractProvider {
+
+        /** For cross-version compatibility. */
+        private static final long serialVersionUID = -7345885830045627291L;
+
+        /** The parameters group. */
+        static final ParameterDescriptorGroup PARAMETERS =
+                createDescriptorGroup(
+                        new NamedIdentifier[] {
```

Comment:
I searched the EPSG database, to my surprise the projection is not recognized/documented. 
However, the name as is seems too generic, as we have 3 variants in the wild
- Non interrupted Homolosine (implemented in proj as "goode", see https://proj.org/operations/projections/goode.html. ESRI does not seem to have it.
- Interrupted Homolosine, land centric, implemented in proj as "igh", see "https://proj.org/operations/projections/igh.html", and called "Goode_Homolosine" by ESRI, choosing "option=1"
- Interrupted Homolosine, water centric, not implemented in proj as far as I can see, and called "Goode_Homolosine" by ESRI, choosing "option=2"

For ESRI references see:
* https://desktop.arcgis.com/en/arcmap/10.3/guide-books/map-projections/goodes-homolosine.htm
* https://github.com/Esri/projection-engine-db-doc/blob/master/text/pe_list_projection.txt#L600
* https://raw.githubusercontent.com/Esri/projection-engine-db-doc/master/text/pe_list_projcs.txt (warning, big file, 6+MB, search for homolosine in it, you'll fine two examples)

Suggestions:

- For ESRI named identifier, use "Goode_Homolosine"
- For GeoTools, maybe "Interrupted_Homolosine" to tell it apart from the non interrupted version?

While now they won't present a big problem, I want to avoid issues in a future where the other two variants are implemented, and make sure we don't end up having conflicts.
The different name for GeoTools tries to be a match with the PROJ convention (calling interrupted what interrupted is), while the land/water difference can be handled by a parameter just like ESRI did, the day someone wants to implement the ocean centric math.